### PR TITLE
editor: adapt code to new widget structure on jsonschema

### PR DESCRIPTION
**Pull request by Garfield-fr**
* Updates `ngx-formly` to 5.12.4.
* Fixes display of reminders in the circulation policy detail view.

Co-Authored-by: Bertrand Zuchuat <bertrand.zuchuat@rero.ch>

## Code review check list

- [ ] Commit message template compliance.
- [ ] Commit message without typos.
- [ ] File names.
- [ ] Functions names.
- [ ] Functions docstrings.
- [ ] Unnecessary commited files?
- [ ] Cypress tests successful?

### CODE REVIEW

**Review by zannkukai — APPROVED** [review]

File: projects/admin/src/app/acquisition/routes/accounts-route.ts [hunk 1]
```diff
@@ -51,8 +51,14 @@ export class AccountsRoute extends BaseRoute implements RouteInterface {
...
             formFieldMap: (field: FormlyFieldConfig, jsonSchema: JSONSchema7): FormlyFieldConfig => {
-              const formOptions = jsonSchema.form;
-              if (formOptions && formOptions.fieldMap === 'amount') {
+              const formWidget = jsonSchema.widget;
+              if (
+                formWidget
+                && formWidget.formlyConfig
+                && formWidget.formlyConfig.templateOptions
+                && formWidget.formlyConfig.templateOptions.fieldMap
+                && formWidget.formlyConfig.templateOptions.fieldMap === 'amount'
```
Comment (jma) [inline projects/admin/src/app/acquisition/routes/accounts-route.ts:60]: can be replaced by `formWidget?.formlyConfig?.templateOptions?.fieldMap`

File: projects/admin/src/app/acquisition/routes/order-lines-route.ts [hunk 2]
```diff
@@ -52,8 +52,14 @@ export class OrderLinesRoute extends BaseRoute implements RouteInterface {
             preCreateRecord: (data: any) => this._addDefaultInformation(data),
             redirectUrl: (record: any) => this.redirectUrl(record.metadata.acq_order, '/records/acq_orders/detail'),
             formFieldMap: (field: FormlyFieldConfig, jsonSchema: JSONSchema7): FormlyFieldConfig => {
-              const formOptions = jsonSchema.form;
-              if (formOptions && formOptions.fieldMap === 'amount') {
+              const formWidget = jsonSchema.widget;
+              if (
+                formWidget
```
Comment (jma) [inline projects/admin/src/app/acquisition/routes/order-lines-route.ts:57]: same as previous

File: projects/admin/src/app/acquisition/routes/receipt-lines-route.ts [hunk 3]
```diff
@@ -49,10 +49,17 @@ export class ReceiptLinesRoute extends BaseRoute implements RouteInterface {
...
             redirectUrl: (record: any) => this.redirectUrl(record.metadata.acq_receipt, '/records/acq_receipts/detail'),
             formFieldMap: (field: FormlyFieldConfig, jsonSchema: JSONSchema7): FormlyFieldConfig => {
-              const formOptions = jsonSchema.form;
-              return (formOptions && formOptions.fieldMap === 'amount')
-                ? this._amountSymbol(field)
-                : field;
+              const formWidget = jsonSchema.widget;
+              if (
+                formWidget
+                && formWidget.formlyConfig
```
Comment (jma) [inline projects/admin/src/app/acquisition/routes/receipt-lines-route.ts:55]: same again

File: projects/admin/src/app/record/custom-editor/user-id-editor/user-id-editor.component.ts [hunk 4]
```diff
@@ -82,15 +85,13 @@ export class UserIdEditorComponent implements OnInit {
       tap(
         schema => {
           if (schema != null) {
+            schema = formToWidget(schema.schema, this._loggerService);
             this.fields = [
-              this._formlyJsonschema.toFieldConfig(orderedJsonSchema(schema.schema), {
+              this._formlyJsonschema.toFieldConfig(orderedJsonSchema(schema), {
```
Comment (jma) [inline projects/admin/src/app/record/custom-editor/user-id-editor/user-id-editor.component.ts:90]: Do we need to call `formToWidget` as `initialRequired` has been added?

File: projects/admin/src/app/routes/circulation-policies-route.ts [hunk 5]
```diff
@@ -80,18 +80,25 @@ export class CirculationPoliciesRoute extends BaseRoute implements RouteInterfac
...
-              }
-              if (formOptions && formOptions.fieldMap === 'amount') {
-                return this._amountSymbol(field);
-              }
-              if (formOptions && formOptions.fieldMap === 'fee_amount') {
-                return this._feeAmountSymbol(field);
+              const formWidget = jsonSchema.widget;
+              if (
+                formWidget
+                && formWidget.formlyConfig
```
Comment (jma) [inline projects/admin/src/app/routes/circulation-policies-route.ts:86]: same as previous

File: projects/admin/src/app/routes/circulation-policies-route.ts [hunk 6]
```diff
@@ -80,18 +80,25 @@ export class CirculationPoliciesRoute extends BaseRoute implements RouteInterfac
...
+                && formWidget.formlyConfig.templateOptions
+                && formWidget.formlyConfig.templateOptions.fieldMap
+              ) {
+                if (formWidget.formlyConfig.templateOptions.fieldMap === 'library') {
```
Comment (jma) [inline projects/admin/src/app/routes/circulation-policies-route.ts:90]: you can use a `switch` here.

File: projects/admin/src/app/routes/collections-route.ts [hunk 7]
```diff
@@ -85,9 +85,14 @@ export class CollectionsRoute extends BaseRoute implements RouteInterface {
               return data;
             },
             formFieldMap: (field: FormlyFieldConfig, jsonSchema: JSONSchema7): FormlyFieldConfig => {
-              const formOptions = jsonSchema.form;
-              if (formOptions && formOptions.hasOwnProperty('fieldMap')) {
-                switch (formOptions.fieldMap) {
+              const formWidget = jsonSchema.widget;
+              if (
+                formWidget
```
Comment (jma) [inline projects/admin/src/app/routes/collections-route.ts:90]: same as previous check for other cases

**Review by jma — CHANGES_REQUESTED** [review]

File: projects/admin/src/app/record/custom-editor/user-id-editor/user-id-editor.component.ts (hunk 4 above)
Comment (Garfield-fr) [reply projects/admin/src/app/record/custom-editor/user-id-editor/user-id-editor.component.ts:90]: This is the case 2 lines above

**Review by jma — APPROVED** [review]

**Review by zannkukai — APPROVED** [review]

**Review by lauren-d — APPROVED** [review]

File: package.json [hunk 8]
```diff
@@ -77,13 +77,13 @@
...
-    "@ngx-formly/bootstrap": "^5.12.1",
-    "@ngx-formly/core": "^5.12.1",
+    "@ngx-formly/bootstrap": "^5.12.4",
+    "@ngx-formly/core": "^5.12.4",
     "@ngx-loading-bar/core": "^5.1.1",
     "@ngx-loading-bar/http-client": "^5.1.1",
     "@ngx-loading-bar/router": "^5.1.1",
     "@ngx-translate/core": "^13.0.0",
-    "@rero/ng-core": "^1.14.6",
+    "@rero/ng-core": "^0.15.0",
```
Comment (PascalRepond) [inline package.json:86]: That's `1.15.0`!

**Review by PascalRepond — CHANGES_REQUESTED** [review]

File: package.json (hunk 8 above)
Comment (Garfield-fr) [reply package.json]: done. Thanks 👍🏻

**Review by PascalRepond — APPROVED** [review]

**Pull request merged**